vcd/resource_vcd_org_user.go

```diff
@@ -0,0 +1,352 @@
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func resourceVcdOrgUser() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdOrgUserCreate,
+		Read:   resourceVcdOrgUserRead,
+		Delete: resourceVcdOrgUserDelete,
+		Update: resourceVcdOrgUserUpdate,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdOrgUserImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateCase("lower"),
+				Description:  "User's name. Only lowercase letters allowed. Cannot be changed after creation",
+			},
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "Organization this user belongs to",
+			},
+			"role": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    false,
+				Description: "Role within the organization",
+			},
+			"password": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+
+				ConflictsWith: []string{"password_file"},
+				Description: "The user's password. This value is never returned on read. " +
+					`Either "password" or "password_file" must be included on creation.`,
+			},
+			"password_file": &schema.Schema{
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      false,
+				ConflictsWith: []string{"password"},
+				Description: "Name of a file containing the user's password. " +
+					`Either "password_file" or "password" must be included on creation.`,
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's description",
+			},
+			"provider_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  govcd.OrgUserProviderIntegrated,
+				ForceNew: false,
+				Description: "Identity provider type for this this user. One of: 'INTEGRATED', 'SAML', 'OAUTH'. " +
+					"When empty, the default value 'INTEGRATED' is used.",
+			},
+			"full_name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's full name",
+			},
+			"email_address": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's email address",
+			},
+			"telephone": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's telephone",
+			},
+			"instant_messaging": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    false,
+				Description: "The Org User's telephone",
+			},
+			"is_enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
```

Comment:
We had this somewhere else and decided that `is_enabled` should be `true` by default and user can explicitly disable with `is_enabled="false"` when needed. This translates to one parameter less to worry about for the easy use cases.

Reply:
That case was different. We took that decision because vCD would set that property (field `advanced` in Edge Gateway) to `true` by default in 9.7+, while leaving the Boolean field out would translate to `false`. 
In this case, the default is always `false`.
I feel more comfortable mimicking vCD behavior.

Reply:
That's a good catch. We did agree to use `enabled`. The only occurrence `is_enabled` is in the old Org resource. @dataclouder let's change to `enabled`.

Reply:
> In this case, the default is always false.
> I feel more comfortable mimicking vCD behavior.

We are defaulting all other enabled fields to true in Terraform. So, having this set to false is setting inconsistent expectations from the user. Let's make it true by default like in other places and people will have one less field to worry about.

Reply:
done